nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py

```diff
@@ -20,16 +22,21 @@
 INSTALL_MESSAGE = "Please install lxml to use this interface!"
 
 
-def subset_shank_channels(recording_extractor: se.RecordingExtractor, xml_file_path: str) -> se.SubRecordingExtractor:
+def subset_shank_channels(
+    recording_extractor: Union[se.RecordingExtractor, si.BaseRecording], xml_file_path: str
+) -> si.BaseRecording:
     """Attempt to create a SubRecordingExtractor containing only channels related to neural data."""
+    if isinstance(recording_extractor, se.RecordingExtractor):
+        recording = si.core.old_api_utils.OldToNewRecording(recording_extractor)
+    else:
+        recording = recording_extractor
     shank_channels = get_shank_channels(xml_file_path=xml_file_path)
+
     if shank_channels is not None:
-        sub_recording = se.SubRecordingExtractor(
-            parent_recording=recording_extractor,
-            channel_ids=[channel_id for group in shank_channels for channel_id in group],
-        )
+        channel_ids = [channel_id for group in shank_channels for channel_id in group]
+        sub_recording = recording.channel_slice(channel_ids)
     else:
-        sub_recording = recording_extractor
+        sub_recording = recording
     return sub_recording
 
```

Comment:
As per @alejoe91 suggestion the subset_shank_channels method in the neuroscope interface has been modified to transform the recorder and use the channel_slice method instead of subrecording.